spec/app/domain/authentication/authn-gcp/extract_resource_restrictions_spec.rb

```diff
@@ -0,0 +1,440 @@
+            extraction_prefix: empty_prefix
+          )
+        end
+
+        it "returns expected list" do
+          expect(subject).to eq(expected_resource_restrictions_list_for_empty_prefix)
+        end
+      end
+
+      context "when prefix case-sensitive with value authn-GCP/" do
+        subject do
+          Authentication::AuthnGCP::ExtractResourceRestrictions.new(
+            resource_class: mocked_resource_class_return_annotations_list
+          ).call(
+            account:           valid_account,
+            username:          valid_host,
+            extraction_prefix: case_sensitive_prefix
+          )
+        end
+
+        it "returns expected list" do
+          expect(subject).to eq(expected_resource_restrictions_list_for_case_sensitive_prefix)
+        end
+      end
+
+      context "when prefix without_slash_prefix" do
+        subject do
+          Authentication::AuthnGCP::ExtractResourceRestrictions.new(
+            resource_class: mocked_resource_class_return_annotations_list
+          ).call(
+            account:           valid_account,
+            username:          valid_host,
+            extraction_prefix: without_slash_prefix
+          )
+        end
+
+        it "returns expected list" do
+          expect(subject).to eq(expected_resource_restrictions_list_for_without_slash_prefix)
+        end
+      end
+
+      context "when prefix is valid with value auth-gcp/" do
+        subject do
+          Authentication::AuthnGCP::ExtractResourceRestrictions.new(
+            resource_class: mocked_resource_class_return_annotations_list
+          ).call(
+            account:           valid_account,
+            username:          valid_host,
+            extraction_prefix: valid_prefix
+          )
+        end
+
+        it "returns expected list" do
+          expect(subject).to eq(expected_resource_restrictions_list_for_valid_prefix)
+        end
+      end
+    end
+
+    context "host not contains annotations" do
+      context "when annotations returned is nil" do
+        subject do
+          Authentication::AuthnGCP::ExtractResourceRestrictions.new(
+            resource_class: mocked_resource_class_return_annotations_nil
+          ).call(
+            account:           valid_account,
+            username:          valid_host,
+            extraction_prefix: valid_prefix
+          )
+        end
+
+        it "raises an error" do
+          expect {subject}.to raise_error(Errors::Conjur::FetchAnnotationsFailed)
+        end
+      end
+
+      context "when annotations returned is empty list" do
+        subject do
+          Authentication::AuthnGCP::ExtractResourceRestrictions.new(
+            resource_class: mocked_resource_class_return_annotations_empty_list
+          ).call(
+            account:           valid_account,
+            username:          valid_host,
+            extraction_prefix: valid_prefix
+          )
+        end
+
+        it "returns expected list" do
```

Comment:
we suppose to fail if we dont have annotations , no?

Reply:
Not it this class, here we only fetch the data